src/consumption.cpp

```diff
@@ -215,191 +227,157 @@ morale_type player::allergy_type( const item &food ) const
     return MORALE_NULL;
 }
 
-bool player::is_allergic( const item &food ) const
+edible_rating player::can_eat( const item &food, std::string *err ) const
```

Comment:
The interface would be cleaner if the return value included the error - return by parameter is C-ish.
For example, if you returned it as a `const std::pair<edible_rating, std::string> &`.